lib/Conch/Route/Device.pm

```diff
@@ -148,6 +117,273 @@ __END__
 
 =pod
 
+Unless otherwise noted, all routes require authentication.
+
+=head3 C<POST /device/:device_id>
+
+=over 4
+
+=item * Request: device_report.yaml
+
+=item * Response: response.yaml#/ValidationStateWithResults
+
+=back
+
+=head3 C<GET /device?:key=:value>
+
+Supports the following query parameters:
+
+=over 4
+
+=item * C</device?hostname=:hostname>
+
+=item * C</device?mac=:macaddr>
+
+=item * C</device?ipaddr=:ipaddr>
+
+=item * C</device?:setting_key=:setting_value>
+
+=back
+
+The value of C<:setting_key> and C<:setting_value> are a device setting key and
+value. For information on how to create a setting key or set its value see
+below.
+
+=over 4
+
+=item * Response: response.yaml#/Devices
+
+=back
+
+=head3 C<GET /device/:device_id>
+
+=over 4
+
+=item * Response: response.yaml#/DetailedDevice
+
+=back
+
+=head3 C<GET /device/:device_id/pxe>
+
+=over 4
+
+=item * Response: response.yaml#/DevicePXE
+
+=back
+
+=head3 C<GET /device/:device_id/phase>
+
+=over 4
+
+=item * Response: response.yaml#/DevicePhase
+
+=back
+
+=head3 C<POST /device/:device_id/graduate>
+
+=over 4
+
+=item * Request: input.yaml#/Null
+
+=item * Response: Redirect to the updated device
+
+=back
+
+=head3 C<POST /device/:device_id/triton_setup>
+
+=over 4
+
+=item * Request: input.yaml#/Null
+
+=item * Response: Redirect to the updated device
+
+=back
+
+=head3 C<POST /device/:device_id/triton_uuid>
+
+=over 4
+
+=item * Request: input.yaml#/DeviceTritonUuid
+
+=item * Response: Redirect to the updated device
+
+=back
+
+=head3 C<POST /device/:device_id/triton_reboot>
+
+=over 4
+
+=item * Request: input.yaml#/Null
+
+=item * Response: Redirect to the updated device
+
+=back
+
+=head3 C<POST /device/:device_id/asset_tag>
+
+=over 4
+
+=item * Request: input.yaml#/DeviceAssetTag
+
+=item * Response: Redirect to the updated device
+
+=back
+
+=head3 C<POST /device/:device_id/validated>
+
+=over 4
+
+=item * Request: input.yaml#/Null
+
+=item * Response: Redirect to the updated device
+
+=back
+
+=head3 C<POST /device/:device_id/phase>
+
+=over 4
+
+=item * Request: input.yaml#/DevicePhase
+
+=item * Response: Redirect to the updated device
+
+=back
+
+=head3 C<GET /device/:device_id/location>
+
+=over 4
+
+=item * Response: response.yaml#/DeviceLocation
+
+=back
+
+=head3 C<POST /device/:device_id/location>
+
+=over 4
+
+=item * Request: input.yaml#/DeviceLocationUpdate
+
+=item * Response: Redirect to the updated device
+
+=back
+
+=head3 C<DELETE /device/:device_id/location>
+
+=over 4
+
+=item * Response: C<204 NO CONTENT>
+
+=back
+
+=head3 C<GET /device/:device_id/settings>
+
+=over 4
+
+=item * Response: response.yaml#/DeviceSettings
+
+=back
+
+=head3 C<POST /device/:device_id/settings>
+
+=over 4
+
+=item * Requires R/W Device Authentication
```

Comment:
What does this mean?

Reply:
Expanded the `R/W` to `Read/Write`, but I'm wondering if we shouldn't have a block in the index about the kinds of permissions in the system.